Creator only

amychan331 opened the pull request:
@typpo For issue #45

typpo commented on imageflow/models.py:
```diff
@@ -174,6 +175,15 @@ def get_short_name(self, maxlen=30):
         n_1 = maxlen - n_2 - 3
         return '%s...%s' % (name[:n_1], name[-n_2:])
 
+    def save(self, user, *args, **kwargs):
```
I think it would actually make sense to approach this problem different way.  I don't think we should overload the `save()` method because:

1) It is a very core method and a lot of existing code relies on its typical method signature (eg. Django plugins and Astrokit code outside the views).  It's quite possible future developers could forget to add the parameter to their save calls.
2) We'd have to overload the method for other editable objects too, like `Reduction` and `LightCurve`
3) This only could prevent unauthorized writes, not unauthorized reads (I know that was not a requirement for this project, but it explains why this is not a typical approach to authorization as far as I know)

Because there is already a `user` attribute in this model, I think it would be better to just check the user when we fetch the light curve in the view logic, not the model logic.  I'll comment on the relevant code below.

typpo commented on imageflow/views.py:
```diff
@@ -396,7 +396,7 @@ def notes(request, pk):
     # TODO(ian): Verify owner of reduction for all these ImageAnalysis fetches.
     analysis = get_object_or_404(ImageAnalysis, pk=pk)
```
To check the user, you could write
```
analysis = get_object_or_404(ImageAnalysis, pk=pk, user=request.user)
```

amychan331 commented:
@typpo I am sorry that I had misunderstood the issue. Since there were both get_object_or_404() and ModelName.objects.get() being used, I thought the issues was created because you wanted a solution that works with both method calls, and so I went with changing the model class for LightCurve and ImageAnalysis. Did not expect so many side-effect - sorry about that! I have removed save() methods and added get_object_or_404(). I changed most of the ones that seems to modify existing model, but I am not sure I got all of them because I wasn’t familiar with how ImageAnalysis is used. Please take a look to see if I miss any ones.

I also added if statements so edit and upload buttons only appears for users who are the LightCurve creator. Should I prevent non-creator from accessing the upload image path in cases where they are not the LightCurve creator?
